AcademicReward/Database/ShopItemDatabase.cs

```diff
@@ -0,0 +1,168 @@
+﻿using AcademicReward.Resources;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using AcademicReward.ModelClass;
+using Npgsql;
+using AcademicReward.Logic;
+using System.Collections.ObjectModel;
+/**
+*  Primary Author: Sean Stille
+*  Reviewer: TBD
+*  Desc: The database implementation for items within the shop, adding them, deleting them, etc.
+*/
+namespace AcademicReward.Database
+{
+    internal class ShopItemDatabase : AcademicRewardsDatabase, IDatabase
+    {
+        ShopLogic logic;
+        public ShopItemDatabase(ShopLogic ShopLogic)
+        {
+            logic = ShopLogic;
+        }
+        public DatabaseErrorType AddItem(object obj)
+        {
+            DatabaseErrorType dbError = DatabaseErrorType.NoError;
+            try
+            {
```

Comment:
Just a thought here and something we can fix later, what if something were to go wrong with your database call? Sure, you are catching the exceptions but always returning that there wasn't an error?

Reply:
In general right now, ShopItemDatabase catches exceptions, preventing a SQL INSERT, in the future I can add more distinct error codes to send back to ShopLogic